x/programs/examples/storage/storage.go

```diff
@@ -0,0 +1,222 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package storage
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	smath "github.com/ava-labs/avalanchego/utils/math"
+	"github.com/ava-labs/hypersdk/consts"
+	"github.com/ava-labs/hypersdk/crypto/ed25519"
+	"github.com/ava-labs/hypersdk/state"
+)
+
+const (
+	HRP           = "simulator_key_"
+	programPrefix = 0x0
+	balancePrefix = 0x1
+
+	BalanceChunks uint16 = 1
+)
+
+var (
+	ErrInvalidBalance = errors.New("invalid balance")
+	balanceKeyPool    = sync.Pool{
+		New: func() any {
+			return make([]byte, 1+ed25519.PublicKeyLen+consts.IDLen+consts.Uint16Len)
+		},
+	}
+)
+
+//
+// Program
+//
+
+func ProgramKey(id ids.ID) (k []byte) {
+	k = make([]byte, 1+consts.IDLen)
+	// convert uint64 to bytes
+	k[0] = programPrefix
+	copy(k[1:], id[:])
+	return
+}
+
+// [programID] -> [programBytes]
+func GetProgram(
+	db state.Immutable,
+	programID ids.ID,
+) (
+	[]byte, // program bytes
+	bool, // exists
+	error,
+) {
+	k := ProgramKey(programID)
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
why not make this an argument of `GetProgram`? Nonetheless, the current cancel usage doesn't make sense to me.